### PR TITLE
implements last saved by feature as a shareable component and adds me…

### PR DESCRIPTION
**Pull request by theseyi**
…tadata to ownership view

### CODE REVIEW

File: wherehows-web/app/components/datasets/containers/dataset-ownership.ts [hunk 1]
```diff
@@ -108,6 +108,29 @@ export default class DatasetOwnershipContainer extends Component {
...
+   */
+  @computed('owners.[]')
+  get ownershipMetadata(): Record<'actor' | 'modificationTime', string> {
+    const {
+      owners: [owner]
+    } = this;
+    const ownershipMetadata = { actor: '', modificationTime: '' };
+
+    if (owner) {
+      const { confirmedBy, modifiedTime } = owner;
```
Comment (cptran777) [inline wherehows-web/app/components/datasets/containers/dataset-ownership.ts]: For owners, whenever someone adds an owner the record that is saved has `confirmedBy` == `UI` so for this situation the metadata would always show "Last saved by UI at <time>". Is that what we're going for?

Comment (theseyi) [reply wherehows-web/app/components/datasets/containers/dataset-ownership.ts]: Wait what? Why was it changed to UI? It used to be the currently logged in user @cptran777

File: wherehows-web/app/components/last-saved-by.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+   * @type {string}
+   * @memberof LastSavedBy
+   */
+  title: string;
+
+  constructor() {
+    super(...arguments);
+
+    // defaults
+    (this.actor && typeof this.actor === 'string') || (this.actor = 'unknown');
```
Comment (igbopie) [inline wherehows-web/app/components/last-saved-by.ts]: Should we create a decorator for this?

File: wherehows-web/app/components/datasets/containers/dataset-ownership.ts (hunk 1 above)
Comment (cptran777) [reply wherehows-web/app/components/datasets/containers/dataset-ownership.ts]: Investigated. Looks like this is an issue on the backend. In our POST request to /owners we definitely send the current user in the `confirmedBy` field, but in the followup GET response the server returns "UI" in the `confirmedBy` field.
Maybe @alyiwang can help investigate, or I can take a look at it later to see what's up.

File: wherehows-web/app/components/last-saved-by.ts (hunk 2 above)
Comment (theseyi) [reply wherehows-web/app/components/last-saved-by.ts]: There is already a decorator [@argument](https://github.com/ember-decorators/argument) but it isn't compatible ([issue here](https://github.com/ember-decorators/argument/issues/65)) with TypeScript at the moment
However, once this emberjs [RFC](https://github.com/emberjs/rfcs/blob/master/text/0337-native-class-constructor-update.md) is implemented, it won't be needed

**Comment by theseyi** [comment]
@cptran777 @igbopie updated PR

**Comment by theseyi** [comment]
Will merge this soon, if not further comments @cptran777 @igbopie

**Pull request merged**